open_learning_ai_tutor/StratL.py

```diff
@@ -250,30 +250,26 @@ def _single_message_tutor(
         tools,
     )
     model = client.model_name
-    assessor_client = options.get("assessor_client", None)
-    assessor = Assessor(
-        client=assessor_client,
-        assessment_history=assessment_history,
-        new_messages=new_messages,
+    tutor = Tutor(
+        client,
+        tools=tools,
     )
+    assesment_prompt = get_assesment_prompt(
+        problem, problem_set, assessment_history, new_messages
+    )
+    assessment_response = tutor.get_response(assesment_prompt)
+    new_assessment_history = assessment_response["messages"]
 
     intermediary = Intermediary.GraphIntermediary2(
         model,
-        assessor=assessor,
+        assessment_history=new_assessment_history,
         intent_history=intent_history,
         chat_history=chat_history,
     )
-    tutor = Tutor.GraphTutor2(
-        client,
-        problem=problem,
-        problem_set=problem_set,
-        model=model,
-        intermediary=intermediary,
-        options=options,
-        tools=tools,
-    )
 
-    new_history, new_intent, new_assessment_history, metadata = tutor.get_response2()
+    prompt, new_intent, metadata = intermediary.get_prompt2(problem, problem_set)
```

Comment:
can we at this point rename get_prompt2 to get_prompt?

Reply:
I was planning to address this in the next pr but could go ahead and do it now